pkg/providerconfig/types/types.go

```diff
@@ -221,7 +220,7 @@ func (configVarString *ConfigVarString) UnmarshalJSON(b []byte) error {
 }
 
 type ConfigVarBool struct {
-	Value           bool                       `json:"value,omitempty"`
+	Value           *bool                      `json:"value,omitempty"`
```

Comment:
I personally really don't like this approach. It lead to this clunky (bool, bool, error) function signature which then the middle boolean is proceeded to be ignored.

The way it's done in the stdlib (and not only there) is much more pleasant, to my taste.

Reply:
I tried with a `Valid bool` approach, but that didn't work out well with the marshalling and unmarshalling tests. Maybe I misunderstood your suggestion.

IMHO, the function signature isn't _that_ clunky. It's additional information that can be used if needed, it just happens to be not needed in most calls (since the provider config parsing doesn't care if the value is `false` because it's set or unset; `false` is what it defaults to anyway).